lib/Sema/SemaResolveUniqueName.cpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include "soll/AST/AST.h"
+#include "soll/AST/Decl.h"
+#include "soll/Sema/Sema.h"
+#include <vector>
+
+namespace soll {
+class DeclUniqueNameResolver : public DeclVisitor {
+  std::vector<Decl *> Stk;
+
+public:
+  DeclUniqueNameResolver() {}
+  void visit(SourceUnitType &SU) override {
+    SU.setUniqueName(
+        "solidity"); // TODO: need different names for multi-sources
```

Comment:
Does Multi-Sources mean the multiple input files or the multiple contracts in one file?